### PR TITLE
Add a placeholder for devdocs

### PR DESCRIPTION
**Pull request by yashrsharma44**
Signed-off-by: Yash <yashrsharma44@gmail.com>

* [ ] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Changes

This PR adds a placeholder for writing up a dev-docs from scratch. Some content has been written up here in a [doc](https://docs.google.com/document/d/1hluKPDKsSdOqcCpOnzr7-4PxLnUtrCl5d4-TVGch_D4/edit)

### CODE REVIEW

File: docs/contributing/onboarding-dev.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: Onboarding Dev Docs
```
Comment (bwplotka) [inline docs/contributing/onboarding-dev.md:2]: It's documentation, so let's use full words here: `Developer Onboarding` maybe?

File: docs/contributing/onboarding-dev.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+- [Store Gateway](#store-gateway)
+- [Querier](#querier)
+- [Ruler](#ruler)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
+
+
+# Onboarding Dev Docs
+This document would help new developers who want to get up and running with Thanos for contributing a technical PR. It contains explanation of different components of Thanos and how they work internally. Each of the components are explained thoroughly and it is expected that after reading this document, the user would be equipped with conceptual clarity to understand and question the internal working of Thanos.
```
Comment (bwplotka) [inline docs/contributing/onboarding-dev.md:24]:
```suggestion

This document would help new developers who want to get up and running with Thanos for contributing a technical PR. It contains explanation of different components of Thanos and how they work internally. Each of the components are explained thoroughly and it is expected that after reading this document, the user would be equipped with conceptual clarity to understand and question the internal working of Thanos.
```

Comment (bwplotka) [reply docs/contributing/onboarding-dev.md:24]: Markdown required new lines betwen those things (:

File: docs/contributing/onboarding-dev.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- [Step by step flow of command for Thanos](#step-by-step-flow-of-command-for-thanos)
+- [Sidecar](#sidecar)
+- [Compact](#compact)
+- [Bucket](#bucket)
+- [Store Gateway](#store-gateway)
+- [Querier](#querier)
+- [Ruler](#ruler)
+
+<small><i><a href='http://ecotrust-canada.github.io/markdown-toc/'>Table of contents generated with markdown-toc</a></i></small>
+
```
Comment (bwplotka) [inline docs/contributing/onboarding-dev.md:20]: Do we need so many new lines? Let's aim for one maybe (:

File: docs/contributing/onboarding-dev.md (hunk 1 above)
Comment (yashrsharma44) [reply docs/contributing/onboarding-dev.md:2]: Sure

**Pull request merged**